main.cpp

```diff
@@ -42,7 +42,139 @@ send me a DM to check your pull request
  */
 
 #include <iostream>
+
+struct FloatType
+{
+    float add( float lhs, float rhs );
+    float subtract( float lhs, float rhs );
+    float multiply( float lhs, float rhs );
+    float divide( float lhs, float rhs );
+};
+
+float FloatType::add( float lhs, float rhs)
+{
+    return lhs + rhs;
+}
+
+float FloatType::subtract( float lhs, float rhs)
+{
+    return lhs - rhs;
+}
+
+float FloatType::multiply( float lhs, float rhs)
+{
+    return lhs * rhs;
+}
+
+float FloatType::divide( float lhs, float rhs)
+{
+    if (rhs == 0 )
+    {
+        std::cout << "Warning! Float division by 0" << std::endl;
+        return lhs / rhs;
+    }
+    return lhs / rhs;
+}
+
+struct IntType
+{
+    int add( int lhs, int rhs );
+    int subtract( int lhs, int rhs );
+    int multiply( int lhs, int rhs );
+    int divide( int lhs, int rhs );
+};
+
+int IntType::add(int lhs, int rhs)
+{
+    return lhs + rhs;
+}
+
+int IntType::subtract(int lhs, int rhs)
+{
+    return lhs - rhs;
+}
+
+int IntType::multiply(int lhs, int rhs)
+{
+    return lhs * rhs;
+}
+
+int IntType::divide(int lhs, int rhs)
+{
+    if ( rhs == 0 )
+    {
+        std::cout << "Warning! Int Division by 0 not possible!" << std::endl;
+        return 0;
+    }
+    return lhs / rhs;  
+}
+
+struct DoubleType
+{
+    double add( double lhs, double rhs );
+    double subtract( double lhs, double rhs );
+    double multiply( double lhs, double rhs );
+    double divide( double lhs, double rhs );
+};
+
+double DoubleType::add(double lhs, double rhs)
+{
+    return lhs + rhs;
+}
+
+double DoubleType::subtract(double lhs, double rhs)
+{
+    return lhs - rhs;
+}
+
+double DoubleType::multiply(double lhs, double rhs)
+{
+    return lhs * rhs;
+}
+
+double DoubleType::divide(double lhs, double rhs)
+{
+    if (rhs == 0 )
```

Comment:
You need to do an `epsilon` comparison here to suppress the warning.
you basically compare if the absolute difference between your `rhs` and some other number is smaller than some epsilon, then that means your `rhs` is basically 0.
- [ ] revise and add epsilon comparison.